### PR TITLE
FEATURE: Add Not Found Page

### PR DESCRIPTION
**Pull request by Yura33-dev**
## Description

This PR introduces 404 page

## Task Link

[Figma Design](https://www.figma.com/design/YKZFEj97V6fe8LADQR5lcq/ONE-FALL?node-id=0-1&t=KIcGZoJcYNdj0cfJ-1)

## Changes Made
- Added `RibbonSvg` UI component
- Laid out `NotFoundError` screen

### CODE REVIEW

File: src/app/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { ArrowLink } from '@/components/ui/ArrowLink/ArrowLink'
+
+import styles from './not-found.module.css'
+
+export default function NotFound() {
+  return (
+    <>
+      <section className={styles.root}>
+        <div className={styles.mainWrapper}>
+          <span className={styles.badge}>404 error</span>
```
Comment (naumch1k) [inline src/app/not-found.tsx]: @Yura33-dev 👋 This element here looks more like a `h1`, while _"We couldn’t find the page you’re looking for"_ can be `h2` or `p`

File: src/app/not-found.module.css [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  overflow: hidden;
+  width: 130vw;
+  height: 42px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-family: var(--font-anton);
+  border: 1px solid #100e11;
+  background-color: #f0cd30;
+  color: #242424;
```
Comment (naumch1k) [inline src/app/not-found.module.css]: @Yura33-dev Please use our defined color variables wherever possible. If there is no such color, please use the RGB format.

File: src/app/not-found.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+              reloading the page. Or go back to home page to find other good
+              stuff.
+            </p>
+            <ArrowLink href='/' target='_self' text='Go to Homepage' />
+          </div>
+        </div>
+      </section>
+
+      <div className={styles.decorWrapper}>
+        <div className={styles.decorLine1}>
```
Comment (naumch1k) [inline src/app/not-found.tsx]: @Yura33-dev Wouldn't that be easier to export those ribbons as SVGs instead of adding that much markup and also styles?

**Review by naumch1k — CHANGES_REQUESTED** [review]
Hey @Yura33-dev, thanks for submitting a pull request!

I'll have to ask you to revisit it as your implementation does not quite match a design (see attached videos).

https://github.com/user-attachments/assets/4c396033-c1b7-4935-83ec-cee9d477849f

https://github.com/user-attachments/assets/b667f0b2-f4cd-4b58-83fb-fb6d113d7451

https://github.com/user-attachments/assets/21665309-ea19-45cb-bef4-f92bbef70d03

- Please make sure section takes all available device view height.
- Whatever way to implement ribbons you choose, please keep them inside section and position them absolutely relative to section. Please, use absolute positioning on all screen widths - not desktop only.
- Please make sure that on smaller screens, the text is positioned above other blocks.

File: src/app/not-found.tsx (hunk 3 above)
Comment (Yura33-dev) [reply src/app/not-found.tsx]: You are absolutely right! 😅

**Comment by Yura33-dev** [comment]
@naumch1k Hey! There is a new commit with updates. Could you check it and please let me know if something else for improvements? Thanks!

**Comment by naumch1k** [comment]
@Yura33-dev 👋 Great job, thanks!

I am adding more changes and would appreciate your review. I split my changes into a few separate commits to make navigating through what has changed easier. Please let me now what you think!

1. I moved the `NotFoundError` component out of the `/app` folder.
2. I refactored `RibbonSvg` UI component to return one element at a time and added an optional `className` prop. I only left size and color styles by `RibbonSvg`; positioning is now handled in `NotFoundError.module.css` - a section where those elements are used. I found that a thinner ribbon looks great on all devices 😍, so I removed another SVG.
3. I ensured the section never exceeded the device screen's height and that there was no scrolling by setting the root to `100dvh`.

https://github.com/user-attachments/assets/9b2defe5-276a-4553-ac9a-ef48cb25c4b7

https://github.com/user-attachments/assets/31e024d9-0443-474d-8759-8cf65407a5c6

https://github.com/user-attachments/assets/af406a4a-ba83-4357-b1c4-56934ab897a6

4. I added a max-width limit - `var(--max-content-width)` - for screens wider than `1280px`.

https://github.com/user-attachments/assets/4a1b9688-3243-4d80-901e-83c913a780aa

5. I applied `align-items: flex-start;` to the flex container with a link to prevent unintended stretching and hover effects when hovering outside its visible area.

https://github.com/user-attachments/assets/4deea83c-5866-4631-9c0c-08d4f32f421c

**Comment by Yura33-dev** [comment]
@naumch1k, Hi! Everything looks fine!

**Review by naumch1k — APPROVED** [review]

**Pull request merged**